### PR TITLE
Aerogear 7524 filtering

### PR DESCRIPTION
**Pull request by cfoskin**
Jira: https://issues.jboss.org/browse/AEROGEAR-7524

This should filter the dataSources by name. will start to work after 3 letters inputted by the user.

It should still list all data sources when no filter is used.

To create some test data : 

```mutation create {
  createDataSource(name: "TestDataSource", type: Postgres, config: "config string: string") {
    id
    name
    type
    config
  }
}
```

### CODE REVIEW

File: ui/components/common/CommonToolbar.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
 import { Toolbar, Button } from "patternfly-react";
+import { DebounceInput } from "react-debounce-input";
 
-const CommonToolbar = ({ buttons }) => {
+const CommonToolbar = ({ buttons, update }) => {
```
Comment (pb82) [inline ui/components/common/CommonToolbar.js]: `update` is a bit generic, can we name it `onFilter` instead?

File: ui/components/data-sources/DataSourcesContainer.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ class DataSourcesContainer extends Component {
         super(props);
         this.state = {
             showModal: false,
-            modalText: null
+            modalText: null,
+            filter: {}
```
Comment (pb82) [inline ui/components/data-sources/DataSourcesContainer.js]: Why an object and not just a string?

Comment (cfoskin) [reply ui/components/data-sources/DataSourcesContainer.js]: The <Query> was failing with an error due to expecting an object and was getting a string. Changing it to an object solved the problem.

Comment (cfoskin) [reply ui/components/data-sources/DataSourcesContainer.js]: The variables object could also have many variables so it expects an object containing these

**Comment by cfoskin** [comment]
@pb82  ok to merge?

**Pull request merged**